src/kong_backend/tests/common/icp_ledger.rs

```diff
@@ -38,37 +43,166 @@ pub struct ArchiveOptions {
+
+    // 2. Read the Wasm module (ICP Ledger Wasm)
+    // ICP_LEDGER_WASM points to "wasm/ledger-canister.wasm.gz"
+    let wasm_module = fs::read(ICP_LEDGER_WASM)?;
+
+    // 3. Encode the init arguments (ICP specific)
+    // Convert to LedgerCanisterPayload as expected by the Candid definition
+    let payload = match ledger_arg {
+        LedgerArg::Init(init_args) => LedgerCanisterPayload::Init(init_args.clone()),
+        LedgerArg::Upgrade(upgrade_args) => LedgerCanisterPayload::Upgrade(Some(upgrade_args.clone())),
+    };
+    
+    // Encode the payload
+    let args = encode_one(payload)?;
+
+    // 4. Install the Wasm onto the specified canister ID
+    ic.install_canister(ledger_canister_id, wasm_module, args, Some(controller));
+
+    Ok(ledger_canister_id)
+}
+
+/// Gets the balance of an ICRC1 account in ICP ledger
+pub fn get_icp_balance(ic: &PocketIc, ledger_id: Principal, account: Account) -> Nat {
+    let payload = encode_one(account).expect("Failed to encode account for balance_of");
+    let response = ic
+        .query_call(ledger_id, Principal::anonymous(), "icrc1_balance_of", payload)
+        .expect("Failed to call icrc1_balance_of on ICP ledger");
+    decode_one::<Nat>(&response).expect("Failed to decode icrc1_balance_of response from ICP ledger")
+}
+
+/// Checks the allowance for an ICRC2 account in ICP ledger
+pub fn get_icp_allowance(
+    ic: &PocketIc,
+    ledger_id: Principal,
+    owner_account: Account,
+    spender_account: Account,
+) -> Allowance {
+    let args = AllowanceArgs {
+        account: owner_account,
+        spender: spender_account,
+    };
+    let payload = encode_one(args).expect("Failed to encode allowance args");
+    let response = ic
+        .query_call(ledger_id, Principal::anonymous(), "icrc2_allowance", payload)
+        .expect("Failed to call icrc2_allowance on ICP ledger");
+    decode_one::<Allowance>(&response).expect("Failed to decode icrc2_allowance response from ICP ledger")
+}
+
+/// Transfers ICRC1 tokens from ICP ledger
+pub fn icp_transfer(
+    ic: &PocketIc, 
+    ledger_id: Principal, 
+    sender: Principal, 
+    to: Account, 
+    amount: Nat, 
+    fee: Option<Nat>, 
+    memo: Option<Vec<u8>>
+) -> Result<Nat, TransferError> {
+    let transfer_args = TransferArg {
+        from_subaccount: None,
+        to,
+        amount,
+        fee,
+        memo: memo.map(|v| icrc_ledger_types::icrc1::transfer::Memo(serde_bytes::ByteBuf::from(v))),
+        created_at_time: None,
+    };
+    let transfer_payload = encode_one(transfer_args).expect("Failed to encode transfer_args");
+    let transfer_response = ic
+        .update_call(ledger_id, sender, "icrc1_transfer", transfer_payload)
+        .expect("Failed to call icrc1_transfer on ICP ledger");
+    decode_one::<Result<Nat, TransferError>>(&transfer_response)
+        .expect("Failed to decode icrc1_transfer response from ICP ledger")
+}
+
+/// Approves an ICRC2 spender in ICP ledger
```

Comment:
```suggestion
#[allow(clippy:too_many_arguments)]
```

do you use clippy? it messes my screens when I get warnings